src/rpcclient/tests/test_client.py

```diff
@@ -19,19 +19,59 @@ def test_poke(client):
         client.poke(peekable, b'a' * 0x100)
 
 
+@pytest.mark.parametrize('params', [
+    ([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]),
+    ([1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15])
+])
+def test_16args(client, params):
+    """
+    :param rpcclient.client.Client client:
+    """
+    with client.safe_malloc(0x100) as peekable:
+        client.symbols.test_16args(peekable, *params)
+        for i in range(len(params)):
+            assert peekable[i] == params[i]
+
+
+@pytest.mark.parametrize('va_list_index,params,expected', [
+    (2, ['%f', 15.5], '15.500000'),
+    (2, ['%f %f', 15.5, 7.3], '15.500000 7.300000'),
+    (2, ['%f %f %d', 15.5, 7.3, 42], '15.500000 7.300000 42'),
+    (2, ['%f %f %d %d', 15.5, 7.3, 1, 2], '15.500000 7.300000 1 2'),
+    (2, ['%f %f %d %d %s', 15.5, 7.3, 1, 2, 'test'], '15.500000 7.300000 1 2 test'),
+    (2, ['%f %f %d %d %s %s', 15.5, 7.3, 1, 2, 'test', 'test2'], '15.500000 7.300000 1 2 test test2'),
+])
+@pytest.mark.arm
+def test_va_list_call(client, va_list_index, params, expected):
+    """
+    :param rpcclient.client.Client client:
+    """
+    with client.safe_malloc(0x100) as peekable:
+        client.symbols.sprintf(peekable, *params, va_list_index=va_list_index)
+        assert expected == peekable.peek_str()
+
+
+@pytest.mark.arm
+def test_floating_point_call(client):
+    """
+    :param rpcclient.client.Client client:
+    """
+    assert client.symbols.sqrt(16.0, return_float64=True) == 4.0
+
+
 def test_peek_invalid_address(client):
     """
     :param rpcclient.client.Client client:
     """
-    with pytest.raises(ArgumentError):
+    with pytest.raises((ArgumentError, ConnectionError)):
```

Comment:
It relies on the server configuration. If compiled with the `-DSAFE_WRITE` flag, attempting to access an invalid address will result in an `ArgumentError`. Otherwise, it will lead to a `ConnectionError`.